hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java

```diff
@@ -41,6 +44,8 @@ public class DefaultClientRegistry implements ClientRegistry {
       "trace.entity.write.excluded.entity.types";
 
   private static final String USER_AGENT_PARSER_CONFIG_KEY = "useragent.parser";
+  private static final String RXJAVA_MAX_IO_THREADS_PROPERTY = "rxjava.threads.io.max";
+  private static final int DEFAULT_RXJAVA_MAX_IO_THREADS = 10;
```

Comment:
I'd suggest giving a lot more than this by default. Since `async.executors.maxPoolSize: 16` by default, we'd want to make sure several IO threads are available per executor I think.